openmc/plotter.py

```diff
@@ -0,0 +1,910 @@
+from numbers import Integral, Real
+from itertools import chain
+import string
+
+import numpy as np
+
+import openmc.checkvalue as cv
+import openmc.data
+
+# Supported keywords for continuous-energy cross section plotting
+PLOT_TYPES = ['total', 'scatter', 'elastic', 'inelastic', 'fission',
+              'absorption', 'capture', 'nu-fission', 'nu-scatter', 'unity',
+              'slowing-down power', 'damage']
+
+# Supported keywoards for multi-group cross section plotting
+PLOT_TYPES_MGXS = ['total', 'absorption', 'scatter', 'fission',
+                   'kappa-fission', 'nu-fission', 'prompt-nu-fission',
+                   'deleyed-nu-fission', 'chi', 'chi-prompt', 'chi-delayed',
+                   'inverse-velocity', 'beta', 'decay rate', 'unity']
+# Create a dictionary which can be used to convert PLOT_TYPES_MGXS to the
+# openmc.XSdata attribute name needed to access the data
+_PLOT_MGXS_ATTR = {line: line.replace(' ', '_').replace('-', '_')
+                   for line in PLOT_TYPES_MGXS}
+_PLOT_MGXS_ATTR['scatter'] = 'scatter_matrix'
+
+# Special MT values
+UNITY_MT = -1
+XI_MT = -2
+
+# MTs to combine to generate associated plot_types
+_INELASTIC = [mt for mt in openmc.data.SUM_RULES[3] if mt != 27]
```

Comment:
Optional: Clarifying comment about importance of `27` would increase readability here (is absorption special)?